### ISSUE TITLE
FE | Profile | [Markup and meta data] HTML markup isn't valid. (09.01.2)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#46799]
#### VFS team name
Profile Team

#### VFS product name
Profile

#### VFS feature name
Bad Address Indicator

#### Point of Contact/Reviewers
Brian DeConinck (@briandeconinck) - Accessibility

*For more information on how to interpret this ticket, please refer to the [Anatomy of a Staging Review issue ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Anatomy-of-a-Staging-Review-Issue-ticket.2060320997.html) guidance on Platform Website.

---

### Platform Issue
HTML markup isn't valid.

### Issue Details
In the "Review your mailing address" alerts, the H2 heading has a `role="alert"` added to it. This isn't a role that's typically allowed to be added to a heading, and it may interfere with the findability of the heading for screen reader users. The `role="alert"` should be placed on a different element.

### Link, screenshot or steps to recreate
The "Review your mailing address" alerts are currently coded as:

```
<h2 slot="headline" tabindex="0" role="alert" aria-live="polite" aria-label="The address we have on file for you may not be correct.">Review your mailing address</h2>
<p>The mailing address we have on file for you may not be correct.</p>
<p><a href="...">...</a></p>
```

### VA.gov Experience Standard
[Category Number 09, Issue Number 01](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Other References
WCAG SC 1.3.1_A

---

### Platform Recommendation
Headings (H1 through H6) have an implicit [heading role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/heading_role). That role is used by screen readers to identify what the headings are and announce them to users.

When you add a different role attribute to a heading... I'm not totally sure what will happen! Testing with NVDA, these H2s are still announced as headings when you encounter them in the page content. But when you tell NVDA to pull up a list of headings on the page, these H2s aren't included on the list. Something weird is happening with the semantics.

I think the intention with the `role="alert"` is to make sure the warning about the bad mailing address is announced to screen reader users on page load, and that does happen. But it might be worth testing it with the `role="alert"` added to a different element. If you're not able to add it directly to the `va-alert` component, maybe try adding it to a `div` containing the `va-alert` component.

### VFS Team Tasks to Complete

- [x] Comment on the ticket if there are questions or concerns
- [x] Close the ticket when the issue has been resolved or validated by your Product Owner. If a team has additional questions or needs Platform help validating the issue, please comment in the ticket.

**Comment by adamwhitlock1** [comment]
I have added the role=alert to the va-alert component instead of the heading, and it is deployed to staging.

@SKasimow, not sure how you would want to validate, but here is a snapshot of the html showing where it is now.

![Screen Shot 2022-09-21 at 9.52.53 AM.png](https://images.zenhubusercontent.com/60ddebd504ce2452d0e0fa92/dae2e1bb-34e1-4691-952d-062d3b9dceab)
